### PR TITLE
[DCP]Add unit test for tp checkpoint 

### PR DESCRIPTION
**Pull request by wz337**
As title.

### CODE REVIEW

File: test/distributed/checkpoint/test_tp_checkpoint.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+from torch.testing._internal.distributed._tensor.common_dtensor import (
+    DTensorTestBase,
+    skip_if_lt_x_gpu,
+    with_comms,
+)
+from torch.testing._internal.distributed.checkpoint_utils import with_temp_dir
+
+
+class ToyModel(nn.Module):
```
Comment (wanchaol) [inline test/distributed/checkpoint/test_tp_checkpoint.py:27]: nit: we can possibly reuse the same model defined in https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/distributed/_tensor/common_dtensor.py#L53

**Review by wanchaol — APPROVED** [review]
LGTM. As a follow up, can you also add some examples in this file? https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/examples/checkpoint_example.py#L136

File: test/distributed/checkpoint/test_tp_checkpoint.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    @with_comms
+    @skip_if_lt_x_gpu(2)
+    @with_temp_dir
+    def test_tp_checkpoint(self):
+        CHECKPOINT_DIR = self.temp_dir
+        mesh_shpe = (self.world_size,)
+        tp_mesh = init_device_mesh(self.device_type, mesh_shpe)
+
+        # create model and move it to GPU with id rank
+        model = ToyModel().cuda(self.rank)
```
Comment (fduwjj) [inline test/distributed/checkpoint/test_tp_checkpoint.py:48]: Can we add a test case where we use Meta device?

**Review by fduwjj — APPROVED** [review]

File: test/distributed/checkpoint/test_tp_checkpoint.py (hunk 2 above)
Comment (wz337) [reply test/distributed/checkpoint/test_tp_checkpoint.py]: Will add that in follow up PR.

**Pull request closed without merging**